### PR TITLE
Tencent COS document loader

### PR DESCRIPTION
**Pull request by Martin7-1**
Hi! I make some progress to integrate with [Tencent COS](https://www.tencentcloud.com/document/product/436) becuase I have some documents on it to load. The `TencentCosDocumentLoaderIT` works correctly in my local, but it need Tencent credentials so I disable it.

### CODE REVIEW

File: document-loaders/langchain4j-document-loader-tencent-cos/pom.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    <dependencies>
+        <dependency>
+            <groupId>dev.langchain4j</groupId>
+            <artifactId>langchain4j-core</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>dev.langchain4j</groupId>
+            <artifactId>langchain4j</artifactId>
```
Comment (langchain4j) [inline document-loaders/langchain4j-document-loader-tencent-cos/pom.xml]: `<scope>test</scope>` is missing, please also move it below to test dependencies

File: langchain4j-bom/pom.xml [hunk 2]
```diff
@@ -169,6 +169,34 @@
...
+
+            <dependency>
+                <groupId>dev.langchain4j</groupId>
+                <artifactId>langchain4j-document-loader-tencent-cos</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>dev.langchain4j</groupId>
+                <artifactId>langchain4j-document-loader-amazon-s3</artifactId>
```
Comment (langchain4j) [inline langchain4j-bom/pom.xml:182]: thank you!

**Review by ghost — COMMENTED** [review]
@Martin7-1 thanks a lot! Please check my comment and we are ready to merge.

File: document-loaders/langchain4j-document-loader-tencent-cos/pom.xml (hunk 1 above)
Comment (Martin7-1) [reply document-loaders/langchain4j-document-loader-tencent-cos/pom.xml]: done

**Comment by Martin7-1** [comment]
Github Action failed because `cos_api` license is not acceptable, but in [Tencent COS sdk repo](https://github.com/tencentyun/cos-java-sdk-v5?tab=MIT-1-ov-file) it seems that it have `MIT` license. @langchain4j Do you know how to solve it? I'm not familiar with license problem.

**Comment by langchain4j** [comment]
@Martin7-1 please add this to `langchain4j-parent/pom.xml`, it should help:
```
<acceptableLicenses>
...
<license>
    <name>cos-java-sdk</name>
    <url>https://github.com/tencentyun/cos-java-sdk-v5</url>
</license>
</acceptableLicenses>
```
Taken from here: https://github.com/tencentyun/cos-java-sdk-v5/blob/master/pom.xml

**Review by ghost — DISMISSED** [review]
@Martin7-1 great job!

**Comment by Martin7-1** [comment]
> @Martin7-1 please add this to `langchain4j-parent/pom.xml`, it should help:
> 
> ```
> <acceptableLicenses>
> ...
> <license>
>     <name>cos-java-sdk</name>
>     <url>https://github.com/tencentyun/cos-java-sdk-v5</url>
> </license>
> </acceptableLicenses>
> ```
> 
> Taken from here: https://github.com/tencentyun/cos-java-sdk-v5/blob/master/pom.xml

@langchain4j Thanks! It works. Everything work fine now.

**Review by ghost — DISMISSED** [review]

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**